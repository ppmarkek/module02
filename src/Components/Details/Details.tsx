import { Results } from '../../types';
import '../Search/style.css';

type DetailsProps = {
  detailsInfo?: Results;
  onClickClose?: () => void;
};

export const Details = ({ detailsInfo, onClickClose }: DetailsProps) => {
  return (
    <div className={'RightCart'}>
      <div className={'Cart'}>
        <h3 data-testid="name">Name: {detailsInfo?.name}</h3>
        <p>Created: {detailsInfo?.createdAt}</p>
        <p>Edited: {detailsInfo?.updatedAt}</p>
        <p>Height: {detailsInfo?.height}</p>
        <p>Mass: {detailsInfo?.mass}</p>
      </div>
      <button data-testid="close-icon" id={'close'} onClick={onClickClose}>
        X
      </button>
    </div>
  );
};
