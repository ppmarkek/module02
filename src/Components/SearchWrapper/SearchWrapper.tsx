import React, { ReactNode } from "react";
import "../Search/style.css";

type SearchWrapperProps = {
  children: ReactNode;
  inputValue: string;
  itemsPage: string;
  newSearch: () => void;
  setItemsPage: (value: string) => void;
  setInputValue?: (value: string) => void;
};

const throwError = () => {
  throw new Error("Intentional error for testing ErrorBoundary");
};

export const SearchWrapper = ({
  children,
  inputValue,
  itemsPage,
  newSearch,
  setItemsPage,
  setInputValue,
}: SearchWrapperProps) => {
  return (
    <div className="wrapper">
      <div className="search">
        <input
          data-testid="search-input"
          onChange={(event) =>
            setInputValue ? setInputValue(event.target.value) : null
          }
          value={inputValue}
        />
        <button
          id="searchButton"
          data-testid="search-button"
          onClick={() => newSearch()}
        >
          Search
        </button>
        <button
          id="errorButton"
          data-testid="error-input"
          onClick={() => throwError()}
        >
          Throw Error
        </button>
        <div className="requestSize">
          <input
            onChange={(event) => setItemsPage(event?.target.value)}
            value={itemsPage}
          />
          <button onClick={() => newSearch()}>Items shown per page</button>
        </div>
      </div>
      {children}
    </div>
  );
};
