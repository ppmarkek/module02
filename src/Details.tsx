import { DetailsType } from './types';
import './App.css';

type DetailsProps = {
  detailsInfo?: DetailsType;
  onClickClose?: () => void;
};

export const Details = ({ detailsInfo, onClickClose }: DetailsProps) => {
  const handleCloseClick = () => {
    if (onClickClose) {
      onClickClose();
    }
  };

  return (
    <div className={'RightCart'}>
      <div className={'Cart'}>
        <h3>Name: {detailsInfo?.name}</h3>
        <p>Created: {detailsInfo?.createdAt}</p>
        <p>Edited: {detailsInfo?.updatedAt}</p>
        <p>Height: {detailsInfo?.height}</p>
        <p>Mass: {detailsInfo?.mass}</p>
      </div>
      <button id={'close'} onClick={handleCloseClick}>
        X
      </button>
    </div>
  );
};
