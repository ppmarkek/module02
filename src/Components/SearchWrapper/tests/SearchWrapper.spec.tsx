import { Details } from "../../Details/Details";
import { fireEvent, render, screen } from "@testing-library/react";
import { resultsMock } from "../../../../fixtures/results";
import React from "react";
import { SearchWrapper } from "../SearchWrapper";

const newSearchEvent = jest.fn();
const setInputValueEvent = jest.fn();
const setItemsPageEvent = jest.fn();

const draw = () => {
  return (
    <SearchWrapper
      inputValue="test input"
      itemsPage="10"
      newSearch={newSearchEvent}
      setItemsPage={setItemsPageEvent}
      setInputValue={setInputValueEvent}
    >
      <>
        <button data-testid="test-button-1" onClick={newSearchEvent}>
          Test Input
        </button>

        <button data-testid="test-button-2" onClick={setItemsPageEvent}>
          Test Search
        </button>

        <button data-testid="test-button-3" onClick={setInputValueEvent}>
          Test Result
        </button>
      </>
    </SearchWrapper>
  );
};

describe("Details Component", () => {
  beforeAll(() => {
    console.error = jest.fn();
    jest.spyOn(console, "error").mockImplementation(() => {});
  });
  it("renders correctly", () => {
    const component = render(draw());

    expect(component).toMatchSnapshot();
  });

  it("calls onClickClose when the close button is clicked", () => {
    const { getByTestId } = render(draw());
    const updateInputButton = getByTestId("test-button-1");
    fireEvent.click(updateInputButton);

    expect(newSearchEvent).toHaveBeenCalledTimes(1);
  });

  it("calls onClickClose when the close button is clicked", () => {
    const { getByTestId } = render(draw());
    const updateInputButton = getByTestId("test-button-2");
    fireEvent.click(updateInputButton);

    expect(setItemsPageEvent).toHaveBeenCalledTimes(1);
  });

  it("calls onClickClose when the close button is clicked", () => {
    const { getByTestId } = render(draw());
    const updateInputButton = getByTestId("test-button-3");
    fireEvent.click(updateInputButton);

    expect(setInputValueEvent).toHaveBeenCalledTimes(1);
  });

  it("calls onClickClose when the close button is clicked", () => {
    const { getByTestId } = render(draw());
    const updateInputButton = getByTestId("error-input");
    fireEvent.click(updateInputButton);

    expect(console.error).toHaveBeenCalled();
  });

  it("calls setInputValue on input change", () => {
    const { getByTestId } = render(draw());
    const input = getByTestId("search-input");
    fireEvent.change(input, { target: { value: "new input" } });

    expect(setInputValueEvent).toHaveBeenCalledWith("new input");
  });
});
