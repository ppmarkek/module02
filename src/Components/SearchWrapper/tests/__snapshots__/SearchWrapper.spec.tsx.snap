// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="wrapper"
      >
        <div
          class="search"
        >
          <input
            data-testid="search-input"
            value="test input"
          />
          <button
            data-testid="search-button"
            id="searchButton"
          >
            Search
          </button>
          <button
            data-testid="error-input"
            id="errorButton"
          >
            Throw Error
          </button>
          <div
            class="requestSize"
          >
            <input
              value="10"
            />
            <button>
              Items shown per page
            </button>
          </div>
        </div>
        <button
          data-testid="test-button-1"
        >
          Test Input
        </button>
        <button
          data-testid="test-button-2"
        >
          Test Search
        </button>
        <button
          data-testid="test-button-3"
        >
          Test Result
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="wrapper"
    >
      <div
        class="search"
      >
        <input
          data-testid="search-input"
          value="test input"
        />
        <button
          data-testid="search-button"
          id="searchButton"
        >
          Search
        </button>
        <button
          data-testid="error-input"
          id="errorButton"
        >
          Throw Error
        </button>
        <div
          class="requestSize"
        >
          <input
            value="10"
          />
          <button>
            Items shown per page
          </button>
        </div>
      </div>
      <button
        data-testid="test-button-1"
      >
        Test Input
      </button>
      <button
        data-testid="test-button-2"
      >
        Test Search
      </button>
      <button
        data-testid="test-button-3"
      >
        Test Result
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
